google-cloud-spanner/acceptance/spanner/client/transaction_test.rb

```diff
@@ -126,6 +128,83 @@
       assert_accounts_equal expected, actual
     end
   end
+  focus
+  it "supports tx isolation with read and update" do
+    results = db.read "accounts", [:reputation], keys: 1, limit: 1
+    original_val = results.rows.first[:reputation]
+    begin
+      latch = Concurrent::CountDownLatch.new 1
+      thr_1 = Thread.new do
+        db.transaction do |tx|
+          tx_results = tx.read "accounts", [:reputation], keys: 1, limit: 1
+          puts "read 1"
+          tx_val = tx_results.rows.first[:reputation]
+          new_val = tx_val + 1
+          tx.update "accounts", [{ account_id: 1, reputation: new_val }]
+          puts "write 1"
+          latch.wait
```

Comment:
Looking at the code here, it seems that update calls commit:
https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-spanner/lib/google/cloud/spanner/session.rb#L433